rllib/agents/alpha_star/alpha_star.py

```diff
@@ -329,7 +332,20 @@ def training_iteration(self) -> ResultDict:
         with self._timers[SYNCH_WORKER_WEIGHTS_TIMER]:
             train_infos = {}
             policy_weights = {}
-            for pol_actor, policy_result in train_results.items():
+            for pol_actor, policy_results in train_results.items():
+                results_have_same_structure = True
+                for result1, result2 in zip(policy_results, policy_results[1:]):
+                    try:
+                        tree.assert_same_structure(result1, result2)
+                    except (ValueError, TypeError):
+                        results_have_same_structure = False
+                        break
+                if len(policy_results) > 1 and results_have_same_structure:
```

Comment:
I need to do something like this for apex as well. right now I'm only getting the most recent info for apex, but I probably need to be averaging the metrics instead.